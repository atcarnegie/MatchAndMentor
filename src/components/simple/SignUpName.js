import React from 'react'
import { View, Text, TextInput, Alert, Keyboard, TouchableWithoutFeedback } from 'react-native'
import Button from './Button'
import ScreenHeader from './ScreenHeader'
import { reduxForm, Field } from 'redux-form'
import { submitName } from '../../actions'
import store from '../../reducers/store'

const styles = {
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'space-between'
  },
  buttonStyle: {
    backgroundColor: '#55A838',
    margin: 20,
    height: 45,
    borderRadius: 10,
    justifyContent: 'center'
  },
  textStyle: {
    color: 'white',
    alignSelf: 'center',
    fontSize: 24
  },
  inputStyle: {
    height: 40,
    bottomBorderColor: 'black',
    bottomBorderWidth: 1,
    margin: 20
  }
}

function submit (values) {
  Keyboard.dismiss()
  if (!values.name) {
    Alert.alert('Submission Error', 'You must enter your name', [{text: 'OK', onPress: () => console.log('OK Pressed')}], { cancelable: false })
  } else {
    store.dispatch(submitName(values.name))
  }
}
function renderInput ({ input: { onChange, ...restInput }, meta: { error } }) {
  return <TextInput style={styles.inputStyle} onChangeText={onChange} {...restInput} maxLength={100}
    onSubmitEditing={Keyboard.dismiss} onEndEditing={Keyboard.dismiss} />
}
function SignUpName (props) {
  return (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
      <View style={styles.container}>
        <View>
          <ScreenHeader title='Your Name' />
          <Field name='name' component={renderInput} />
        </View>
        <View>
          <Button style={styles.buttonStyle} onPress={props.handleSubmit(submit)} testID='name-submit-button'>
            <Text style={styles.textStyle}>Continue</Text>
          </Button>
        </View>
      </View>
    </TouchableWithoutFeedback>
  )
}

export default reduxForm({ form: 'name' })(SignUpName)
