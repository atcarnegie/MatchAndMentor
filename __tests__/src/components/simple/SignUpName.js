/* eslint-env jest */
'use strict'
import { Keyboard } from 'react-native'
import React from 'react'
import { Provider } from 'react-redux'
import SignUpName from '../../../../src/components/simple/SignUpName'
import renderer from 'react-test-renderer'
import store from '../../../../src/reducers/store'
import { findById } from '../../../../setupTests'

jest.mock('TouchableOpacity', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableHighlight')
})

jest.mock('Keyboard', () => {
  return { dismiss: jest.fn() }
})

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}><SignUpName /></Provider>
  )
  expect(tree).toMatchSnapshot()
})

it('calls submit on button press', () => {
  let tree = renderer.create(<Provider store={store}><SignUpName /></Provider>).toJSON()
  const submitButton = findById(tree, 'name-submit-button')
  submitButton.props.onPress()
  expect(Keyboard.dismiss).toBeCalled()
})
