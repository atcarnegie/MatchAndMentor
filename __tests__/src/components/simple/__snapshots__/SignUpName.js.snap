// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
  testID={undefined}
>
  <View>
    <View
      style={
        Object {
          "margin": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 24,
          }
        }
      >
        Your Name
      </Text>
    </View>
    <TextInput
      maxLength={100}
      name="name"
      onBlur={[Function]}
      onChangeText={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onEndEditing={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      style={
        Object {
          "bottomBorderColor": "black",
          "bottomBorderWidth": 1,
          "height": 40,
          "margin": 20,
        }
      }
      value=""
    />
  </View>
  <View>
    <TouchableHighlight
      activeOpacity={0.85}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#55A838",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "margin": 20,
        }
      }
      testID="name-submit-button"
      underlayColor="black"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        Continue
      </Text>
    </TouchableHighlight>
  </View>
</View>
`;
